components/kserve/kserve.go

```diff
@@ -136,18 +136,23 @@ func (k *Kserve) ReconcileComponent(ctx context.Context, cli client.Client,
 		}
 	}
 
+	err = k.configureServiceMesh(cli, dscispec)
+	if err != nil {
+		return err
+	}
```

Comment:
Perhaps we could scope `err` like like we do for other paths here, but most importantly add some more context about where this error occured. Might help troubleshooting in the future.

```suggestion
	if err = k.configureServiceMesh(cli, dscispec); err != nil {
		return fmt.Errorf("failed configuring service mesh while reconciling kserve component. cause: %w", err);
	}
```